src/overlays/actors/ovl_En_Ta/z_en_ta.c

```diff
@@ -107,80 +142,83 @@ void EnTa_Init(Actor* thisx, PlayState* play2) {
+                this->eyeIndex = TALON_EYE_INDEX_CLOSED;
                 Animation_PlayOnce(&this->skelAnime, &gTalonSleepAnim);
                 this->currentAnimation = &gTalonSleepAnim;
                 this->actor.shape.shadowScale = 54.0f;
             }
             break;
-        case 2:
+        case ENTA_RETURNED_FROM_KAKARIKO:
+            // "Return Talon"
             osSyncPrintf(VT_FGCOL(CYAN) " 出戻りタロン \n" VT_RST);
-            if (!GET_EVENTCHKINF(EVENTCHKINF_6B)) {
+            if (!GET_EVENTCHKINF(EVENTCHKINF_TALON_RETURNED_FROM_KAKARIKO)) {
                 Actor_Kill(&this->actor);
             } else if (!LINK_IS_ADULT) {
                 Actor_Kill(&this->actor);
             } else if (play->sceneId == SCENE_MALON_STABLE && !IS_DAY) {
                 Actor_Kill(&this->actor);
                 osSyncPrintf(VT_FGCOL(CYAN) " 夜はいない \n" VT_RST);
             } else {
-                func_80B13AA0(this, func_80B14D98, func_80B167C0);
-                this->eyeIndex = 0;
+                EnTa_SetActionFunctions(this, EnTa_IdleBackAtRanch, EnTa_AnimLoopCurrent);
+                this->eyeIndex = TALON_EYE_INDEX_OPEN;
                 Animation_PlayOnce(&this->skelAnime, &gTalonStandAnim);
                 this->currentAnimation = &gTalonStandAnim;
             }
             break;
         default:
+            // "Other Talon" == child Talon
             osSyncPrintf(VT_FGCOL(CYAN) " その他のタロン \n" VT_RST);
-            if (play->sceneId == SCENE_SPOT15) {
-                if (GET_EVENTCHKINF(EVENTCHKINF_14)) {
+            if (play->sceneId == SCENE_SPOT15) { // Hyrule Castle
+                if (GET_EVENTCHKINF(EVENTCHKINF_TALON_RETURNED_FROM_CASTLE)) {
                     Actor_Kill(&this->actor);
-                } else if (GET_EVENTCHKINF(EVENTCHKINF_13)) {
-                    func_80B13AA0(this, func_80B14C18, func_80B167C0);
-                    this->eyeIndex = 0;
+                } else if (GET_EVENTCHKINF(EVENTCHKINF_TALON_WOKEN_IN_CASTLE)) {
+                    EnTa_SetActionFunctions(this, EnTa_IdleWokenInCastle, EnTa_AnimLoopCurrent);
+                    this->eyeIndex = TALON_EYE_INDEX_OPEN;
                     Animation_PlayOnce(&this->skelAnime, &gTalonStandAnim);
                     this->currentAnimation = &gTalonStandAnim;
                 } else {
-                    func_80B13AA0(this, func_80B14634, func_80B167FC);
-                    this->eyeIndex = 2;
+                    EnTa_SetActionFunctions(this, EnTa_IdleAsleepInCastle, EnTa_AnimSleeping);
+                    this->eyeIndex = TALON_EYE_INDEX_CLOSED;
                     Animation_PlayOnce(&this->skelAnime, &gTalonSleepAnim);
                     this->currentAnimation = &gTalonSleepAnim;
                     this->actor.shape.shadowScale = 54.0f;
                 }
-            } else if (play->sceneId == SCENE_SOUKO) {
+            } else if (play->sceneId == SCENE_SOUKO) { // Talon's house
```

Comment:
We avoid commenting on the scene checks as the plan is to name them something more sensible than the debug names eventually (work started in #1343 )